### PR TITLE
chore: Update Ruma to 684ffc

### PR DESCRIPTION
**Pull request by Hywan**
This patch updates Ruma to the latest commit on its `main` branch.

This is useful for https://github.com/matrix-org/matrix-rust-sdk/issues/2932.

### CODE REVIEW

**Review by richvdh — APPROVED** [review]
lgtm

**Comment by richvdh** [comment]
... except it doesn't pass CI.

I think that updating to the latest version of ruma is going to be a bit more complicated, and we might as well fix https://github.com/matrix-org/matrix-rust-sdk/issues/2932 at the same time rather than spending a bunch of effort attempting to preserve the current, broken, behaviour.

**Comment by Hywan** [comment]
I've forgot to fix a test. Now it should pass. I don't know how to fix the “current broken behaviour”, I can just update to the latest Ruma version for you. My skill in crypto stops here :-).

**Comment by Hywan** [comment]
@richvdh Apart from codecov, everything is alright now. I propose to merge this PR, and continue to fix #2932 in another PR. Thoughts?

File: crates/matrix-sdk-crypto/src/secret_storage.rs
Comment (richvdh) [inline crates/matrix-sdk-crypto/src/secret_storage.rs:254]: For the record, I think the correct way to handle this is something like:

```suggestion
                let Some(iv), Some(mac) = (&properties.iv, &properties.mac) else {
                    // The IV and/or MAC are missing from the account data content. As the [spec] says, we have
                    // to assume that the key is valid.
                    //
                    // [spec]: https://spec.matrix.org/unstable/client-server-api/#msecret_storagev1aes-hmac-sha2
                    return Ok(());
                };
```

However, if you want to land this as-is, I'll not object.

**Review by richvdh — APPROVED** [review]
Thanks!

File: crates/matrix-sdk-crypto/src/secret_storage.rs
Comment (Hywan) [reply crates/matrix-sdk-crypto/src/secret_storage.rs]: I've included your suggestion, thanks! The correct syntax is `let (Some(iv), Some(mac)) = …` though, otherwise I would have accepted it directly from the Github UI.

Waiting for the CI to validate this change before merging.

**Pull request merged**
